src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -1626,15 +1626,15 @@ private void EmitInstanceCallExpression(BoundCall call, UseKind useKind)
                 }
                 else
                 {
-                    // calling a method defined in a base class.
+                    // calling a method defined in a base class or interface.
 
                     // When calling a method that is virtual in metadata on a struct receiver, 
                     // we use a constrained virtual call. If possible, it will skip boxing.
                     if (method.IsMetadataVirtual())
                     {
                         // NB: all methods that a struct could inherit from bases are non-mutating
```

Comment:
>all methods that a struct could inherit from bases are non-mutating [](http://example.com/codeflow?start=31&length=67)

This assumption looks suspicious. For example, ToString when overridden in a struct can mutate the struct.